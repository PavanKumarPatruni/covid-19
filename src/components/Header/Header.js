import React from 'react';

import './Header.scss';

const Header = () => {
  return (
    <div className="full-width comp-mb fadeInUp">
      <h1 className="header flex flex-justify-center py4">COVID-19</h1>
    </div>
  );
};

export default Header;
