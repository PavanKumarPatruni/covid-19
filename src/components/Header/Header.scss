@import '../../styles/index.scss';

.header {
  @include heading($text-color);

  background-color: $background;
  border-bottom: 1px solid $header-shadow;
  box-shadow: 0 2px 10px $header-shadow;

  @include mobile {
    @include heading-mobile($text-color);
  }

  &.light-mode {
    @include heading($text-color-light);

    background-color: $white;
    border-bottom: 1px solid $light-grey;
    box-shadow: 0 2px 10px $light-grey;

    @include mobile {
      @include heading-mobile($text-color-light);
    }
  }
}

.toggle-button {
  right: 16px;
  top: 44px;

  @include mobile {
    top: 28px;
  }

  @include x-small {
    top: 20px;
  }
}
