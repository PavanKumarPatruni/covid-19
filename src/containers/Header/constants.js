export const COVID_CHANGE_COLOR_MODE = 'COVID/COLOR/CHANGE_COLOR_MODE';
