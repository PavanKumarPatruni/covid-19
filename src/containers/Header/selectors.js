import { createSelector } from 'reselect';

const getColorMode = state => {
  return state.colorModeData;
};

export const selectColorMode = createSelector(
  getColorMode,
  data => data.colorMode
);
