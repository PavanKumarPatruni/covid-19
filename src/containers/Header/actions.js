import { COVID_CHANGE_COLOR_MODE } from './constants';

export const changeColorMode = () => {
  return {
    type: COVID_CHANGE_COLOR_MODE,
  };
};
