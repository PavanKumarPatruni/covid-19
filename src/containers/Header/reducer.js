import { COVID_CHANGE_COLOR_MODE } from './constants';

const INITIAL_STATE = {
  colorMode: 'dark',
};

const getColorModeReducer = (state = INITIAL_STATE, action) => {
  const { type } = action;
  switch (type) {
    case COVID_CHANGE_COLOR_MODE:
      return {
        ...state,
        colorMode: state.colorMode === 'white' ? 'dark' : 'white',
      };
    default:
      return state;
  }
};

export { getColorModeReducer };
