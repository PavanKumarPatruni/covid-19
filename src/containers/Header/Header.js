import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import Header from '../../components/Header';

import { changeColorMode } from './actions';
import { selectColorMode } from './selectors';

export const mapDispatchToProps = dispatch => {
  return {
    changeColorMode: () => {
      dispatch(changeColorMode());
    },
  };
};

const mapStateToProps = createStructuredSelector({
  colorMode: selectColorMode,
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
