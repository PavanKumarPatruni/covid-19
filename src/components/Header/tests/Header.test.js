import React from 'react';
import { shallow, mount } from 'enzyme';

import Header from '../Header';

describe('Header component', () => {
  const props = {
    changeColorMode: () => {},
    colorMode: 'dark',
  };

  test('match snapshot correctly', () => {
    const wrapper = shallow(<Header {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('match snapshot correctly', () => {
    props.colorMode = 'light';
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('match snapshot correctly', () => {
    const wrapper = mount(<Header {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
