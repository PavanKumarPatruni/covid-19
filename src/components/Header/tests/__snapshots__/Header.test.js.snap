// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component match snapshot correctly 1`] = `
<div
  className="header py4 full-width relative flex md-flex-justify-center sm-flex-start flex-center dark-mode"
>
  <h1
    className="sm-pl3"
  >
    COVID-19
  </h1>
  <div
    className="absolute toggle-button"
  >
    <ThemeToggle
      onChange={[Function]}
    />
  </div>
</div>
`;

exports[`Header component match snapshot correctly 2`] = `
<div
  className="header py4 full-width relative flex md-flex-justify-center sm-flex-start flex-center light-mode"
>
  <h1
    className="sm-pl3"
  >
    COVID-19
  </h1>
  <div
    className="absolute toggle-button"
  >
    <ThemeToggle
      onChange={[Function]}
    />
  </div>
</div>
`;

exports[`Header component match snapshot correctly 3`] = `
<Header
  changeColorMode={[Function]}
  colorMode="light"
>
  <div
    className="header py4 full-width relative flex md-flex-justify-center sm-flex-start flex-center light-mode"
  >
    <h1
      className="sm-pl3"
    >
      COVID-19
    </h1>
    <div
      className="absolute toggle-button"
    >
      <ThemeToggle
        onChange={[Function]}
      >
        <button
          className="theme-toggle p2 sm-p1"
          onClick={[Function]}
          type="button"
        >
          <ArrowUp
            darkMode={false}
            size={48}
          >
            <svg
              fill="#201c3a"
              height={48}
              stroke="#201c3a"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width={48}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"
              />
            </svg>
          </ArrowUp>
        </button>
      </ThemeToggle>
    </div>
  </div>
</Header>
`;
